### PR TITLE
Celery json serializer

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

File: kitsune/upload/tasks.py [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
 from PIL import Image
 
-
 log = logging.getLogger('k.task')
 
 
 @task(rate_limit='15/m')
 def generate_thumbnail(for_obj, from_field, to_field,
-                       max_size=settings.THUMBNAIL_SIZE):
+                       max_size=settings.THUMBNAIL_SIZE, serializer="pickle"):
```
Comment (LeoMcA) [inline kitsune/upload/tasks.py:18]: Fair enough, could we pass a path to the image instead?

File: kitsune/upload/tasks.py [hunk 2]
```diff
@@ -100,7 +98,7 @@ def _scale_dimensions(width, height, longest_side=settings.THUMBNAIL_SIZE):
 
 
 @task(rate_limit='15/m')
-def compress_image(for_obj, for_field):
+def compress_image(for_obj, for_field, serializer="pickle"):
```
Comment (LeoMcA) [inline kitsune/upload/tasks.py:101]: Same q as above

File: kitsune/customercare/tasks.py [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
 
 
 @task()
-def maybe_award_badge(badge_template, year, user):
+def maybe_award_badge(badge_template: Dict, year: int, user_id: int):
     """Award the specific badge to the user if they've earned it."""
+    try:
+        user = User.objects.get(id=user_id)
+    except User.DoesNotExist:
+        return
```
[the remark this answers is not in the document]
Comment (akatsoulas) [reply kitsune/customercare/tasks.py:18]: Maybe something in between like fail silently but log the error/send it to sentry.

File: kitsune/upload/tasks.py (hunk 1 above)
Comment (akatsoulas) [reply kitsune/upload/tasks.py:18]: Worth investigating

File: kitsune/customercare/tasks.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (LeoMcA) [reply kitsune/customercare/tasks.py:18]: Yeah sending the error to sentry would be good

**Pull request merged**